components/shared/posts/CodeEditor.tsx

```diff
@@ -0,0 +1,181 @@
+"use client";
+import { Textarea } from "@/components/ui/textarea";
+
+import { CodeIcon, Eye } from "lucide-react";
+import React, { useState } from "react";
+
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { ErrorMessage } from "@hookform/error-message";
+import { uploadImage } from "@/lib/actions/cloudinary.actions";
+import { LuUploadCloud } from "react-icons/lu";
+import { toast } from "@/components/ui/use-toast";
+import Image from "next/image";
+interface CodeEditorProps {
+  register: any;
+  watch: any;
+  errors?: any;
+  className?: string;
+  setValue: any;
+  postType?: string;
+}
+const CodeEditor = ({
+  register,
+  watch,
+  errors,
+  className,
+  setValue,
+  postType,
+}: CodeEditorProps) => {
+  const code = watch("code");
+  const watchInput = watch("image");
+  const [uploadedImageUrl, setUploadedImageUrl] = useState<string | null>(
+    watchInput
+  );
+  const inputRef = React.useRef<HTMLInputElement | null>(null);
+  const { ref, ...rest } = register("image");
+
+  const generateLineNumbers = () => {
+    const lines = code.split("\n");
+    return lines.map((_: string, idx: number) => (
+      <div key={idx}>{idx + 1}</div>
+    ));
+  };
+  const readFileAsDataURL = (file: Blob) => {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onloadend = () => resolve(reader.result);
+      reader.onerror = reject;
+      reader.readAsDataURL(file);
+    });
+  };
+
+  const handleFileChange = async (event: any) => {
+    const file = event.target.files[0];
+    if (!file) return;
+
+    const fileData = await readFileAsDataURL(file);
+
+    try {
+      const uploadResult = await uploadImage(fileData, {
+        action: "postCreation",
+      });
+      setUploadedImageUrl(uploadResult);
+      if (uploadResult) {
+        setValue("image", uploadResult);
+      }
+      toast({
+        type: "foreground",
+        variant: "default",
+        title: "Image Uploaded Successfully",
+      });
+    } catch (error) {
+      console.log(error);
+      toast({
+        type: "foreground",
+        variant: "destructive",
+        title: "Error Uploading Image",
+      });
+    }
+  };
+
+  return (
+    <>
+      <Tabs className={` w-full ga space-y-2 text-white-300` + `${className}`}>
```

Comment:
```suggestion
      <Tabs defaultValue="code" className={` w-full ga space-y-2 text-white-300` + `${className}`}>
```
This will fix the 'code' section of the tabs not loading immediately. Giving a default value auto selects one of the tabs.

```suggestion
      <Tabs defaultValue="code" className={` w-full space-y-2 text-white-300 ${className}`}>
```
Not sure what 'ga' is? I would remove it. Also when using template literals- we don't need to use `+` signs.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals